### PR TITLE
Fix most rendering warnings from switching to CommonMark

### PR DESCRIPTION
**Pull request by steveklabnik**
There's one big one lift, I'm filing a bug for it soon.

r? @rust-lang/docs

### CODE REVIEW

**Comment by steveklabnik** [comment]
cc @alexcrichton who noticed this

**Comment by steveklabnik** [comment]
Other bug is https://github.com/rust-lang/rust/issues/45420

**Comment by QuietMisdreavus** [comment]
r=me pending travis. apparently the footnote issue was fixed in a later version of pulldown, so it's just a matter of updating rustdoc to point at a version that contains that fix. I opened https://github.com/rust-lang/rust/pull/45421 to do that.

**Comment by QuietMisdreavus** [comment]
The travis logs confirm that the only warnings left in std are connected to the footnote in libcore's crate-root docs. After double-checking against the latest commit in https://github.com/rust-lang/rust/pull/45421, i'm reasonably sure that that should grab those warnings.

@bors r+ rollup

File: src/libcore/hash/sip.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ use mem;
 /// This is currently the default hashing function used by standard library
 /// (eg. `collections::HashMap` uses it by default).
 ///
-/// See: https://131002.net/siphash/
+/// See: <https://131002.net/siphash>
```
Comment (ishitatsuyuki) [inline src/libcore/hash/sip.rs]: Nitpick: inconsistent trailing slash.

Comment (steveklabnik) [reply src/libcore/hash/sip.rs]: both work

Comment (ollie27) [reply src/libcore/hash/sip.rs]: The version without the trailing slash results in a 301 redirect.

**Pull request merged**